nntrainer/models/neuralnet.cpp

```diff
@@ -187,14 +186,18 @@ static void setLabels(const std::vector<Tensor> &data,
 static void setInputs(const std::vector<Tensor> &data,
                       const std::vector<Var_Grad *> &input_list) {
 
-  NNTR_THROW_IF(data.size() != input_list.size(), std::invalid_argument)
+  NNTR_THROW_IF(data.size() > 1 && data.size() != input_list.size(),
+                std::invalid_argument)
     << "input size does not match with the network requirements"
     << " input size: " << data.size()
     << " requirements size: " << input_list.size();
 
   /// feed or clear label
   for (unsigned int idx = 0; idx < input_list.size(); idx++) {
-    input_list[idx]->initializeVariable(data[idx]);
+    if (data.empty())
+      input_list[idx]->initializeVariable();
+    else
+      input_list[idx]->initializeVariable(data[idx]);
```

Comment:
as now idx is depending on input_list, it looks valid to do

```cpp
  else(data.size() == 1){
   input_list[idx]->initializeVariable(data[0]);
  }
```

Reply:
This is done in the upcoming PR - https://github.com/nnstreamer/nntrainer/pull/1544/files#diff-8136da3afc181f3163a8f8a15d86d132abee8c07de2c9b49bc2cf0759e380a69R833-R845. If you don't mind, I will let it be resolved in the upcoming PR itself.

Reply:
I don't mind :)